### DISCUSSION TITLE
Software Recommendation for System Topology Drawings

### DISCUSSION THREAD

**Discussion by energydatachpguy** [oemof/tespy#590] [General]
Hi Francesco, 

I love the clean topology drawing you include on the advanced tutorials page of tespy documentation, specifically the one for the heat pump. Do you mind sharing a recommendation for what software to use to make those nice drawings? 

Eventually I would like to manually generate those topology SVG files, and populate the different values from your tespy modeling onto the svg for sharing with different clients for plant design cases. 

I mentioned in a separate comment earlier, but thank you so much for the work you and your colleagues do on this project. It is extremly helpful, and I am trying each day to learn more about how to use it. 

Regards
John

**Comment by fwitte** [comment]
Good day @johnfeldhausen and happy new year!

Thank you for your patience, my computer was out of reach for a couple of days.

> I love the clean topology drawing you include on the advanced tutorials page of tespy documentation, specifically the one for the heat pump. Do you mind sharing a recommendation for what software to use to make those nice drawings?

For my drawings I have used [inkscape](https://inkscape.org/). However, it does not automize the process unfortunately. I have not found time to implement a solution... One possibility might be to leverage the capabilities of [draw.io](https://app.diagrams.net/), which can be exported to svg and also json. And you can import json-style data for the drawings, which I think would have good potential.

> I mentioned in a separate comment earlier, but thank you so much for the work you and your colleagues do on this project. It is extremly helpful, and I am trying each day to learn more about how to use it.

Thank you, I am happy to hear that :).

Best

Francesco

**Reply by energydatachpguy** [reply]
Thank you for the guidance on inkscape. Curious if you have a github folder location with the different building block symbols that correspond to the components you use when making your drawings? 

John

**Reply by fwitte** [reply]
Hi John! Find them here: https://github.com/oemof/tespy/tree/dev/docs%2Fapi%2F_images

Best

Francesco

**Reply by PabloMBarral** [reply]
Hi!

Just to mention that using attin or attout, don't remember, function in AutoCad and block attributes you can update the data in the pfd with every calculation. It's quite useful. Values in the pfd illustrate flows much better than tables.

Not quite sure if you can import svg files 
Should be possible, because a pfd can be imported


Sorry for the offtopic.

**Comment by Nika250** [comment]
thanks for the information!
